org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/marker/GradleErrorMarker.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2018 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core.marker;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.List;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+
+import org.eclipse.buildship.core.CorePlugin;
+import org.eclipse.buildship.core.util.string.StringUtils;
+
+import org.eclipse.buildship.core.workspace.GradleBuild;
+
+/**
+ * Describes Gradle error marker.
+ *
+ * @author Donat Csikos
+ */
+public class GradleErrorMarker {
+
+    public static String ID = CorePlugin.PLUGIN_ID + ".errormarker";
+    public static String ATTRIBUTE_STACKTRACE = "stacktrace";
+    public static String ATTRIBUTE_ROOT_DIR = "rootdir";
+
+    private static final String LINE_SEPARATOR = System.getProperty("line.separator");
+
+    private GradleErrorMarker() {
+    }
+
+    public static boolean belongsToBuild(IMarker marker, GradleBuild build) {
+        String rootDir = marker.getAttribute(ATTRIBUTE_ROOT_DIR, null);
+        return build.getBuildConfig().getRootProjectDirectory().getAbsolutePath().equals(rootDir);
+    }
+
+    public static void create(IResource resource, GradleBuild gradleBuild, String message, Throwable exception, int lineNumber) throws CoreException {
+        IMarker marker = resource.createMarker(GradleErrorMarker.ID);
+
+        if (lineNumber >= 0) {
+            marker.setAttribute(IMarker.LINE_NUMBER, lineNumber);
+        }
+
+        marker.setAttribute(IMarker.MESSAGE, collectErrorMessages(exception));
+        marker.setAttribute(IMarker.PRIORITY, IMarker.PRIORITY_HIGH);
+        marker.setAttribute(IMarker.SEVERITY, IMarker.SEVERITY_ERROR);
+        marker.setAttribute(IMarker.LOCATION, "Gradle build script");
```

Comment:
⭕️ It's not necessarily a build script - only if the resource is a file.

Reply:
Any suggestion what should we display here? Maybe nothing?

<img width="936" alt="screen shot 2018-01-08 at 13 45 40" src="https://user-images.githubusercontent.com/419883/34671295-55e76cbc-f47a-11e7-9276-c1bbd7312b58.png">

Reply:
I've removed the location field.

Reply:
Yeah, it doesn't really add value.